lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp

```diff
@@ -203,12 +266,12 @@ namespace Js
 
     Var DynamicObjectPropertyEnumerator::MoveAndGetNext(PropertyId& propertyId, PropertyAttributes * attributes)
     {
+        if (this->cachedData && this->initialType == this->object->GetDynamicType())
```

Comment:
Did this->initialType always use to have the same value as this->cachedDataType?